app/src/main/java/fr/free/nrw/commons/contributions/ContributionBoundaryCallback.kt

```diff
@@ -0,0 +1,90 @@
+package fr.free.nrw.commons.contributions
+
+import androidx.paging.PagedList.BoundaryCallback
+import fr.free.nrw.commons.Media
+import fr.free.nrw.commons.auth.SessionManager
+import fr.free.nrw.commons.di.CommonsApplicationModule
+import fr.free.nrw.commons.media.MediaClient
+import io.reactivex.Scheduler
+import io.reactivex.disposables.CompositeDisposable
+import timber.log.Timber
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Named
+
+/**
+ * Class that extends PagedList.BoundaryCallback for contributions list It defines the action that
+ * is triggered for various boundary conditions in the list
+ */
+class ContributionBoundaryCallback @Inject constructor(
+    private val repository: ContributionsRepository,
+    private val sessionManager: SessionManager,
+    private val mediaClient: MediaClient,
+    @param:Named(CommonsApplicationModule.IO_THREAD) private val ioThreadScheduler: Scheduler
+) : BoundaryCallback<Contribution>() {
+    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
+
+    /**
+     * It is triggered when the list has no items User's Contributions are then fetched from the
+     * network
+     */
+    override fun onZeroItemsLoaded() {
+        fetchContributions()
+    }
+
+    /**
+     * It is triggered when the user scrolls to the top of the list No action is taken at this point
+     */
+    override fun onItemAtFrontLoaded(itemAtFront: Contribution) {}
+
+    /**
+     * It is triggered when the user scrolls to the end of the list User's Contributions are then
+     * fetched from the network
+     */
+    override fun onItemAtEndLoaded(itemAtEnd: Contribution) {
+        fetchContributions()
+    }
+
+    /**
+     * Fetches contributions using the MediaWiki API
+     */
+    fun fetchContributions() {
+        compositeDisposable.add(
+            mediaClient.getMediaListForUser(sessionManager.userName)
+                .map<List<Contribution>> { mediaList: List<Media?> ->
```

Comment:
you can remove the type information from this `map`

also my original point here was we could map the mediaList and avoid the overly verbose loop eg
```
mediaList.map { Contribution(it, Contribution.STATE_COMPLETED) }
```

and the OnNext action in the subscribe can become a method reference

```
.subscribe(
   ::saveContributionsToDb, 
```